integration/README.md

```diff
@@ -7,17 +7,52 @@ These are VM based tests for core u-root functionality such as:
 -   uinit (user init), and
 -   running unit tests requiring root privileges.
 
-To learn more about how these tests work under the hood, see the next section,
-otherwise jump ahead to the sections on how to write and run these tests.
+## Overview
+
+All tests are in the integration/ directory. Within that, there are a
+few subdirectories:
+
+* generic-tests/ : most tests can be put under this.
+* golang-tests/ : this is for Go unit tests that can be run inside the VM.
+* testcmd/ : this contains custom uinits for tests.
+* testdata/ : this contains any extra files for tests.
+
+### Generic Tests
+
+This is where most tests live. main\_test.go is the test orchestrator. It is 
+responsible for running all the tests in the directory.
+
+All tests must have an entry in the tests table in main\_test.go. Each test 
+entry has 2 things: a name, and a runner function. This runner function is where
+the actual test is taking place.
+
+main\_test.go will then go through the table and call the runner function
+for each test.
+
+When main\_test.go is invoked using `go test`, it takes a few flags:
+* kernel: path to the Linux kernel binary to use. eg. `-kernel="$HOME/linux/arch/x86/boot/bzImage"`
+* qemu: path to the QEMU binary and args to use. eg. `-qemu="$HOME/bin/qemu-system-x86_64 -enable-kvm"`
+* [optional] initramfs: path to an initramfs to use for all the tests. If one is not
```

Comment:
From my understanding, this "overrides" whatever initramfs the test would have used. Would it make sense to use the word "overrides" here.